lib/archethic/p2p/message.ex

```diff
@@ -237,1731 +208,26 @@ defmodule Archethic.P2P.Message do
-
-    {offset, rest} = VarInt.get_value(rest)
-
-    {%TransactionInputList{
-       inputs: inputs,
-       more?: more?,
-       offset: offset
-     }, rest}
-  end
-
-  def decode(<<245::8, rest::bitstring>>) do
-    {length, rest} = rest |> VarInt.get_value()
-
-    {%TransactionChainLength{
-       length: length
-     }, rest}
-  end
-
-  def decode(<<246::8, rest::bitstring>>) do
-    {nb_new_seeds, rest} = rest |> VarInt.get_value()
-    {new_seeds, <<rest::bitstring>>} = deserialize_node_list(rest, nb_new_seeds, [])
-
-    {nb_closest_nodes, rest} = rest |> VarInt.get_value()
-    {closest_nodes, rest} = deserialize_node_list(rest, nb_closest_nodes, [])
-
-    {%BootstrappingNodes{
-       new_seeds: new_seeds,
-       closest_nodes: closest_nodes
-     }, rest}
-  end
-
-  def decode(<<247::8, digest_size::8, digest::binary-size(digest_size), rest::bitstring>>) do
-    {%EncryptedStorageNonce{
-       digest: digest
-     }, rest}
-  end
-
-  def decode(<<248::8, uco_balance::64, rest::bitstring>>) do
-    {nb_token_balances, rest} = rest |> VarInt.get_value()
-    {token_balances, rest} = deserialize_token_balances(rest, nb_token_balances, %{})
-
-    {%Balance{
-       uco: uco_balance,
-       token: token_balances
-     }, rest}
-  end
-
-  def decode(<<249::8, rest::bitstring>>) do
-    {nb_nodes, rest} = rest |> VarInt.get_value()
-    {nodes, rest} = deserialize_node_list(rest, nb_nodes, [])
-    {%NodeList{nodes: nodes}, rest}
-  end
-
-  def decode(<<250::8, rest::bitstring>>) do
-    {nb_unspent_outputs, rest} = rest |> VarInt.get_value()
-
-    {unspent_outputs, <<more_bit::1, rest::bitstring>>} =
-      deserialize_versioned_unspent_output_list(rest, nb_unspent_outputs, [])
-
-    more? = more_bit == 1
-
-    {offset, rest} = VarInt.get_value(rest)
-
-    {%UnspentOutputList{unspent_outputs: unspent_outputs, more?: more?, offset: offset}, rest}
-  end
-
-  def decode(<<251::8, rest::bitstring>>) do
-    {nb_transactions, rest} = rest |> VarInt.get_value()
-    {transactions, rest} = deserialize_tx_list(rest, nb_transactions, [])
-
-    case rest do
-      <<0::1, rest::bitstring>> ->
-        {
-          %TransactionList{transactions: transactions, more?: false},
-          rest
-        }
-
-      <<1::1, paging_state_size::8, paging_state::binary-size(paging_state_size),
-        rest::bitstring>> ->
-        {
-          %TransactionList{transactions: transactions, more?: true, paging_state: paging_state},
-          rest
-        }
-    end
-  end
-
-  def decode(<<252::8, rest::bitstring>>) do
-    Transaction.deserialize(rest)
-  end
-
-  def decode(<<253::8, rest::bitstring>>) do
-    {%NotFound{}, rest}
-  end
-
-  def decode(<<254::8, rest::bitstring>>) do
-    {%Ok{}, rest}
+  def decode(<<message_id::8, rest::bitstring>>) do
+    msg_module = MessageId.id_to_module(message_id)
+    msg_module.deserialize(rest)
   end
 
   def decode(<<255::8>>), do: raise("255 message type is reserved for stream EOF")
```

Comment:
This clause should be before the precedent:

```
iex(1)> Archethic.P2P.Message.decode(<<255::8>>)
** (FunctionClauseError) no function clause matching in Archethic.P2P.MessageId.id_to_module/1   
```

Reply:
nice catch